### PR TITLE
MCO move sources for certs used by etcd

### PR DESCRIPTION
**Pull request by abhinavdahiya**
Move the operator to use config maps instead of on disk files for root and etcd-ca
Move etcd static pod to use `/etc/kubernetes`

Requires: https://github.com/openshift/installer/pull/624

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
testing locally with installer from https://github.com/openshift/installer/pull/624
```console
$ oc get nodes
NAME                       STATUS     ROLES     AGE       VERSION
adahiya-0-master-0         Ready      master    5m        v1.11.0+d4cacc0
adahiya-0-worker-0-9zdsm   NotReady   worker    6s        v1.11.0+d4cacc0
```

File: cmd/machine-config-operator/bootstrap.go [hunk 1]
```diff
@@ -36,6 +38,8 @@ var (
 
 func init() {
 	rootCmd.AddCommand(bootstrapCmd)
+	bootstrapCmd.PersistentFlags().StringVar(&bootstrapOpts.etcdCAFile, "etcd-ca", "/etc/ssl/etcd/ca.crt", "path to etcd CA certificate")
+	bootstrapCmd.PersistentFlags().StringVar(&bootstrapOpts.rootCAFile, "root-ca", "/etc/ssl/kubernetes/ca.crt", "path to root CA certificate")
```
Comment (wking) [inline cmd/machine-config-operator/bootstrap.go]: Don't we want to look for these [under `/etc/kubernetes`][1]?

[1]: https://github.com/openshift/installer/issues/622#issuecomment-436317953

Comment (abhinavdahiya) [reply cmd/machine-config-operator/bootstrap.go]: These are default values.
Also MCO runs inside a container so volume mounts take take of most of the stuff.

**Comment by abhinavdahiya** [comment]
https://github.com/openshift/installer/pull/624 merged
/test e2e-aws

**Comment by abhinavdahiya** [comment]
still using old installer
```console
sh-4.2$ oc -n openshift-machine-config-operator get pods
NAME                                       READY     STATUS    RESTARTS   AGE
machine-config-operator-757fff9b67-dknf8   1/1       Running   0          4m
sh-4.2$ oc -n openshift-machine-config-operator logs machine-config-operator-757fff9b67-dknf8
I1106 19:09:38.967861       1 start.go:39] Version: 3.11.0-208-ga2c1960c-dirty
I1106 19:09:39.069255       1 leaderelection.go:185] attempting to acquire leader lease  openshift-machine-config-operator/machine-config-operator...
E1106 19:09:39.480098       1 event.go:259] Could not construct reference to: '&v1.ConfigMap{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"machine-config-operator", GenerateName:"", Namespace:"openshift-machine-config-operator", SelfLink:"/api/v1/namespaces/openshift-machine-config-operator/configmaps/machine-config-operator", UID:"82571a74-e1f7-11e8-a7ba-12d8394c36be", ResourceVersion:"2093", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:63677128179, loc:(*time.Location)(0x1c0be80)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string{"control-plane.alpha.kubernetes.io/leader":"{\"holderIdentity\":\"machine-config-operator-757fff9b67-dknf8_82198acb-e1f7-11e8-8f1a-0a580a02000a\",\"leaseDurationSeconds\":90,\"acquireTime\":\"2018-11-06T19:09:39Z\",\"renewTime\":\"2018-11-06T19:09:39Z\",\"leaderTransitions\":0}"}, OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, Data:map[string]string(nil), BinaryData:map[string][]uint8(nil)}' due to: 'no kind is registered for the type v1.ConfigMap'. Will not report event: 'Normal' 'LeaderElection' 'machine-config-operator-757fff9b67-dknf8_82198acb-e1f7-11e8-8f1a-0a580a02000a became leader'
I1106 19:09:39.480186       1 leaderelection.go:194] successfully acquired lease openshift-machine-config-operator/machine-config-operator
I1106 19:09:39.666602       1 operator.go:144] Starting MachineConfigOperator
E1106 19:09:40.368168       1 reflector.go:205] github.com/openshift/machine-config-operator/pkg/generated/informers/externalversions/factory.go:101: Failed to list *v1.ControllerConfig: the server could not find the requested resource (get controllerconfigs.machineconfiguration.openshift.io)
E1106 19:09:43.266872       1 reflector.go:205] github.com/openshift/machine-config-operator/pkg/generated/informers/externalversions/factory.go:101: Failed to list *v1.ControllerConfig: the server could not find the requested resource (get controllerconfigs.machineconfiguration.openshift.io)
W1106 19:09:44.387045       1 reflector.go:341] github.com/openshift/machine-config-operator/pkg/generated/informers/externalversions/factory.go:101: watch of *v1.ControllerConfig endedwith: too old resource version: 2220 (2221)
E1106 19:09:53.891984       1 operator.go:206] configmaps "root-ca" not found
E1106 19:10:03.492067       1 operator.go:206] configmaps "root-ca" not found
E1106 19:10:14.191055       1 operator.go:206] configmaps "root-ca" not found
E1106 19:10:23.774273       1 operator.go:206] configmaps "root-ca" not found
E1106 19:10:33.180677       1 operator.go:206] configmaps "root-ca" not found
```
/test e2e-aws

File: cmd/machine-config-operator/bootstrap.go (hunk 1 above)
Comment (wking) [reply cmd/machine-config-operator/bootstrap.go]: > These are default values.

Why not match the defaults to paths that are more native?  If we expect this to run from a container (and we do), then `/etc/kubernetes/...` seems like a better fit.  If we expect this to be run on the host (without a container runtime), then `/etc/pki/...` seems like a better fit.  @cgwalters has `/etc/ssl/...` as [a Debian/Gentoo-ism][1].

[1]: https://github.com/openshift/installer/issues/622#issuecomment-436318092

Comment (abhinavdahiya) [reply cmd/machine-config-operator/bootstrap.go]: > Why not match the defaults to paths that are more native? If we expect this to run from a container (and we do), then /etc/kubernetes/... seems like a better fit.

inside the container `/etc/ssl` seem more useful; `/etc/kubernetes` is not required inside the container.

gonna keep the defaults as-is unless required.

**Comment by abhinavdahiya** [comment]
from the ci
```console
oc get nodes,pods --all-namespaces
sh-4.2$ oc get pods -n openshift-machine-config-operator
NAME                                         READY     STATUS    RESTARTS   AGE
machine-config-controller-6cb6c9b554-bqpp4   1/1       Running   0          4m
machine-config-daemon-97sfx                  1/1       Running   0          3m
machine-config-daemon-h9tn5                  1/1       Running   0          3m
machine-config-daemon-q7l5b                  1/1       Running   0          3m
machine-config-operator-67f989c8fd-xhz7h     1/1       Running   0          8m
machine-config-server-mdvdd                  1/1       Running   0          3m
machine-config-server-whznv                  1/1       Running   0          3m
machine-config-server-x777b                  1/1       Running   0          3m
```

all the components of MCO are running.

File: cmd/machine-config-operator/bootstrap.go (hunk 1 above)
Comment (cgwalters) [reply cmd/machine-config-operator/bootstrap.go]: > inside the container /etc/ssl seem more useful;

Yeah, I doubt we have a lot of consistency on this point.  Keeping `/etc/ssl` inside the container seems OK to me too.

**Comment by cgwalters** [comment]
LGTM

**Pull request merged**